src/AmazonS3EncryptionClientBase.cs

```diff
@@ -70,8 +71,8 @@ internal AmazonS3Client S3ClientForInstructionFile
 	    }
 
         internal AmazonS3CryptoConfigurationBase S3CryptoConfig { get; set; }
-        internal readonly Dictionary<string, UploadPartEncryptionContext> CurrentMultiPartUploadKeys = new Dictionary<string, UploadPartEncryptionContext>();
-        internal readonly Dictionary<InitiateMultipartUploadRequest, UploadPartEncryptionContext> AllMultiPartUploadRequestContexts = new Dictionary<InitiateMultipartUploadRequest, UploadPartEncryptionContext>();
+        internal readonly ConcurrentDictionary<string, UploadPartEncryptionContext> CurrentMultiPartUploadKeys = new ConcurrentDictionary<string, UploadPartEncryptionContext>();
```

Comment:
We should use the framework version of ConcurrentDictionary for every platform except .NET 3.5. That should just be a difference of swapping using statements depending on the platform.

Reply:
Agreed, I fixed in revision 2.